streamer/periodconcat_node.py

```diff
@@ -0,0 +1,256 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Concatenates inputs into periods by creating a master DASH/HLS file."""
+
+import os
+import re
+from typing import List, Tuple
+from xml.etree import ElementTree
+from streamer import __version__
+from streamer.node_base import NodeBase, ProcessStatus, ThreadedNodeBase
+from streamer.transcoder_node import TranscoderNode
+from streamer.packager_node import PackagerNode
+from streamer.input_configuration import Input
+from streamer.pipeline_configuration import PipelineConfig, ManifestFormat, StreamingMode
+
+
+class PeriodConcatNode(ThreadedNodeBase):
+  """A node that concatenates multiple DASH manifests and/or HLS playlists
+  when the input is a multiperiod_inputs_list.
+  """
+  
+  def __init__(self,
+               pipeline_config: PipelineConfig,
+               all_nodes: List[NodeBase],
+               outpud_dir: str) -> None:
+    """Stores all relevant information needed for the period concatenation."""
+    super().__init__(thread_name='periodconcat', continue_on_exception=False, sleep_time=3)
+    self._pipeline_config = pipeline_config
+    self._output_dir = outpud_dir
+    
+    self._period_inputs_list: List[List[Input]] = []
+    # When iterating with enumeruate(List[NodeBase]) each single index will point
+    # to one TranscoderNode and its corresponding PackagerNode in each of these lists.
+    self._transcoder_nodes: List[TranscoderNode] = []
+    self._packager_nodes: List[PackagerNode] = []
+    
+    # We don't need to save CloudNode(s) or ExternalCommandNode(s).
+    # We will start the concatenation once all the PackagerNode(s) are finished.
+    for node in all_nodes:
+      if isinstance(node, TranscoderNode):
+        self._transcoder_nodes.append(node)
+        self._period_inputs_list.append(node._inputs)
```

Comment:
It's poor form to access a private member of another class.  Also, I don't see this being used anywhere.

Reply:
I thought i might need it for the HLS, but i will remove it for now.